pkg/data/egress_lists/egress_lists.go

```diff
@@ -31,12 +24,60 @@ var templateGCPClassic string
 //go:embed aws-hcp-zeroegress.yaml
 var templateAWSHCPZeroEgress string
 
-func GetLocalEgressList(platformType cloud.Platform) (string, error) {
-	if !platformType.IsValid() {
-		fmt.Printf("platform type %s is invalid", platformType)
+type githubReposClient interface {
+	GetContents(ctx context.Context, owner, repo, path string, opts *github.RepositoryContentGetOptions) (fileContent *github.RepositoryContent, directoryContent []*github.RepositoryContent, resp *github.Response, err error)
+}
+
+type Generator struct {
+	PlatformType      cloud.Platform
+	Variables         map[string]string
+	logger            logging.Logger
+	githubReposClient githubReposClient
+}
+
+func NewGenerator(platformType cloud.Platform, variables map[string]string, logger logging.Logger) *Generator {
+	return &Generator{
+		PlatformType:      platformType,
+		Variables:         variables,
+		logger:            logger,
+		githubReposClient: github.NewClient(nil).Repositories,
+	}
+}
```

Comment:
I'm curious about utility of a factory method like this for an exported struct. If the intent is provide a framework for type validation, couldn't someone get around this by just instantiating a `Generator` directly?

Reply:
The intent is to provide a common way to create a `Generator` without having to think about the implementation details needed to call GitHub via our newly added interface. If some creates a `Generator` directly outside of this package they will end up getting a panic when trying to get GitHub egress lists (which I could fix for extra safety with error handling). I am open to other suggestions, I just find factory functions easier to reason about what is needed for a struct to work, and provides an easy mechanism to add further input validation in the future.

Reply:
Thanks for the explanation! That makes sense to me